### PR TITLE
Wire up cloud facade to use new service apis

### PR DESCRIPTION
**Pull request by wallyworld**
A previous PR moved update and revoke credential business logic from the apiserver facade layer to the new credential service. This PR wires up that logic into the facade, replacing the now obsolete facade logic which is deleted.

Also, to unblock further logic cleanup, the apiserver credentialcommon logic to validate credentials has the dependencies on state and rpc/params removed. But there's still more to do for the rest of the package. The use of export_test is also removed as we are not wanting to do that any more.

Providers are given a context to use when making cloud apis calls; this context is used to mark a credential as invalid when needed. However, we currently need to update both dqlite and mongo. For this PR, an adaptor is introduced for use with the update / remove credential logic to illustrate the 2 calls needed. But this is short term and another PR will be needed to address the issue properly which will involve churn across many call sites.

Second commit includes a drive by fix for creating the controller api facade - the cloud service was not wired up resulting in a NPE doing model migration pre checks. Also fix model export to include the credential - now model migration works again.

## QA steps

Tested on AWS.
First bootstrap.
`show-model` and `show-credential aws <name> --show-secrets` display the credential and model credential details

Edit the local credential so it is invalid.
`update-credential aws <name>` fails with an error because the credential is invalid.
`update-credential aws <name> --force` succeeds with a warning.

Depending on timing, `show-model` and `show-credential` will display that the credential is invalid (not all provider invalid credential calls update both databases yet).

Edit the local credential so it is valid.
`update-credential aws <name>` updates the credential and it is now displayed as valid.

`remove-credential aws <name>` fails because the credential is in use.

`add-credential aws ...`  to add a second credential for aws.
`remove-credential aws <name2>` works because the credential is not in use.

bootstrap a second controller
migrate a model from controller1->controller2

## Links

**Jira card:** JUJU-4622

### CODE REVIEW

File: domain/credential/state/state.go [hunk 1]
```diff
@@ -356,6 +356,10 @@ func (st *State) CloudCredential(ctx context.Context, name, cloudName, owner str
 		return cloud.Credential{}, errors.Trace(err)
 	}
 
+	if name == "" && cloudName == "" && owner == "" {
```
Comment (hpidcock) [inline domain/credential/state/state.go:359]: Should this be `OR` instead of `AND`?

File: apiserver/facades/client/cloud/cloud.go [hunk 2]
```diff
@@ -445,16 +424,6 @@ func (api *CloudAPI) AddCredentials(ctx context.Context, args params.TaggedCrede
 	return results, nil
 }
 
-// CheckCredentialsModels validates supplied cloud credentials' content against
-// models that currently use these credentials.
-// If there are any models that are using a credential and these models or their
-// cloud instances are not going to be accessible with corresponding credential,
-// there will be detailed validation errors per model.
-// There's no Juju API client which uses this, but JAAS does,
-func (api *CloudAPI) CheckCredentialsModels(ctx context.Context, args params.TaggedCredentials) (params.UpdateCredentialResults, error) {
```
Comment (hpidcock) [inline apiserver/facades/client/cloud/cloud.go:454]: So we confirmed this is no longer used yeah?

**Review by hpidcock — APPROVED** [review]

File: apiserver/facades/client/cloud/cloud.go (hunk 2 above)
Comment (wallyworld) [reply apiserver/facades/client/cloud/cloud.go:454]: Yeah, JAAS guys are aware  - it was just added for them as far as I am aware. It was the old way to update a model credential.

File: domain/credential/state/state.go (hunk 1 above)
Comment (wallyworld) [reply domain/credential/state/state.go:359]: Yeah, fixed. Gone in next PR anyway.

**Pull request merged**